diff -up apcupsd-3.14.8/platforms/redhat/apcupsd.in.init apcupsd-3.14.8/platforms/redhat/apcupsd.in
--- apcupsd-3.14.8/platforms/redhat/apcupsd.in.init	2008-11-01 16:03:26.000000000 +0100
+++ apcupsd-3.14.8/platforms/redhat/apcupsd.in	2011-04-15 12:10:17.352352927 +0200
@@ -3,7 +3,7 @@
 # apcupsd      This shell script takes care of starting and stopping
 #	       the apcupsd UPS monitoring daemon.
 #
-# chkconfig: 2345 60 99
+# chkconfig: - 60 99
 # description: apcupsd monitors power and takes action if necessary
 #
 APCPID=@PIDDIR@/apcupsd.pid
