--- apcupsd-3.14.0/platforms/redhat/apcupsd.in.init	2005-01-09 15:55:41.000000000 -0700
+++ apcupsd-3.14.0/platforms/redhat/apcupsd.in	2007-04-23 12:13:35.000000000 -0600
@@ -3,7 +3,7 @@
 # apcupsd      This shell script takes care of starting and stopping
 #	       the apcupsd UPS monitoring daemon.
 #
-# chkconfig: 2345 60 99
+# chkconfig: - 60 99
 # description: apcupsd monitors power and takes action if necessary
 #
 APCPID=@PIDDIR@/apcupsd.pid
